api/public/public_api/endpoints/wake_word_file.py

```diff
@@ -0,0 +1,169 @@
+    job moves them to a permanent one.  Each file will be logged on the sample table
+    for their location and classification data.
+    """
+
+    _file_location = None
+    _wake_word_repository = None
+    _wake_word = None
+
+    def __init__(self):
+        super(WakeWordFileUpload, self).__init__()
+        self.request_data = None
+
+    @property
+    def wake_word_repository(self):
+        """Lazy instantiation of wake word repository object."""
+        if self._wake_word_repository is None:
+            self._wake_word_repository = WakeWordRepository(self.db)
+
+        return self._wake_word_repository
+
+    @property
+    def wake_word(self):
+        """Build and return a WakeWord object."""
+        if self._wake_word is None:
+            self._wake_word = self.wake_word_repository.ensure_wake_word_exists(
+                name=self.request_data["wake_word"].strip(),
+                engine=self.request_data["engine"],
+            )
+
+        return self._wake_word
+
+    @property
+    def file_location(self):
+        """Build and return a TaggingFileLocation object."""
+        if self._file_location is None:
+            data_dir = Path(environ["SELENE_DATA_DIR"])
+            wake_word = self.request_data["wake_word"].replace(" ", "-")
+            wake_word_dir = data_dir.joinpath("wake-word").joinpath(wake_word)
+            wake_word_dir.mkdir(parents=True, exist_ok=True)
+            file_location_repository = TaggingFileLocationRepository(self.db)
+            self._file_location = file_location_repository.ensure_location_exists(
+                server=LOCAL_IP, directory=str(wake_word_dir)
+            )
+
+        return self._file_location
+
+    def post(self, device_id):
+        """
+        Process a HTTP POST request submitting a wake word sample from a device.
+
+        :param device_id: UUID of the device that originated the request.
+        :return:  HTTP response indicating status of the request.
+        """
+        self._authenticate(device_id)
+        self._validate_post_request()
+        account = self._get_account(device_id)
+        file_name = self._save_audio_file(account)
+        self._add_wake_word_file(account, file_name)
+
+        return jsonify("Wake word sample uploaded successfully"), HTTPStatus.OK
+
+    def _validate_post_request(self):
+        """Load the post request into the validation class and perform validations."""
+        upload_request = UploadRequest(
+            dict(
+                wake_word=self.request.form.get("wake_word"),
+                engine=self.request.form.get("engine"),
+                timestamp=self.request.form.get("timestamp"),
+                model=self.request.form.get("model"),
+            )
+        )
+        upload_request.validate()
+        self.request_data = upload_request.to_native()
+        if "audio_file" not in self.request.files:
+            raise DataError(dict(audio_file="No audio file included in request"))
```

Comment:
Who catches this and what does the user ultimately see? Why not catch this in validate?

Reply:
We have a Flask blueprint used by all our APIs that looks for this exception and returns a "bad request"